### ISSUE TITLE
problems using on Venom Linux with SSL???

### ISSUE THREAD

**Issue by bja-au** [IBBoard/cawbird#460]
Hi - have been trying to get Cawbird running on Venom Linux and get the following messages when starting up ...

(cawbird:1722): dbind-WARNING : 18:56:52.190: Couldn't connect to accessibility bus: Failed to connect to socket /var/run/user/65/at-spi/bus_0: No such file or directory

(cawbird:1722): libsoup-WARNING : 18:56:52.260: Could not set SSL credentials from '/etc/ssl/certs': Failed to populate trust list from /etc/ssl/certs: Error while reading file.

(cawbird:1722): libsoup-WARNING : 18:56:52.260: Could not set SSL credentials from '/etc/ssl/certs': Failed to populate trust list from /etc/ssl/certs: Error while reading file.

(cawbird:1722): Gtk-WARNING : 18:56:52.356: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.

(cawbird:1722): libsoup-WARNING : 18:56:52.389: Could not set SSL credentials from '/etc/ssl/certs': Failed to populate trust list from /etc/ssl/certs: Error while reading file.

(cawbird:1722): libsoup-WARNING : 18:56:52.389: Could not set SSL credentials from '/etc/ssl/certs': Failed to populate trust list from /etc/ssl/certs: Error while reading file.

.. and ever time it tries to display a pic I get ...

(cawbird:6617): cawbird-WARNING **: 23:52:49.695: Couldn't load pixbuf: Request on 'https://pbs.twimg.com/media/FWjqOyPXkAA6vmX.jpg:small' returned status 'SSL handshake failed' (https://pbs.twimg.com/media/FWjqOyPXkAA6vmX.jpg:small)

.... If I goto th URL in my browser it displays fine????

Any thoughts please??

Thanks
BJA

**Comment by CodedOre** [comment]
Cawbird uses libsoup to make connections, and it seems it can't create an secure connection with SSL to the Twitter servers, as it has troubles opening that file.

Since Venom Linux is unknown to me: How do you run Cawbird? Do you use a package from the distribution, the Flatpak or did you build it from source?

**Comment by bja-au** [comment]
Hi - thanks for your reply.

Venom Linux is a non-systemd distro (see venomlinux.org). I have libsoup installed. The following is the openssl info ...

-su-5.1# openssl version -a
OpenSSL 1.1.1o  3 May 2022
built on: Fri Jun 24 07:36:50 2022 UTC
platform: linux-x86_64
options:  bn(64,64) rc4(16x,int) des(int) idea(int) blowfish(ptr) 
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -march=x86-64 -pipe -Os -fomit-frame-pointer -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DZLIB -DZLIB_SHARED -DNDEBUG
OPENSSLDIR: "/etc/ssl"
ENGINESDIR: "/usr/lib/engines-1.1"
Seeding source: os-specific
-su-5.1# 
... Cawbird was built from source however I also used a version from devuan linux which produced the same errors.
Regards
BJA

**Comment by IBBoard** [comment]
The key bit is `Could not set SSL credentials from '/etc/ssl/certs': Failed to populate trust list from /etc/ssl/certs: Error while reading file.` - where does Venom Linux store its SSL certificates? Wherever it is, libsoup isn't configured to access them properly. But that's out of our control, because we don't set anything (because each distro is different).

If your certificates aren't in `/etc/ssl/certs` then it *should* work to make it a symlink to the right location. Mine on openSUSE is a symlink to `/var/lib/ca-certificates/pem/` as a standard system thing.

**Comment by CodedOre** [comment]
I've checked Venom Linux inside a Virtual Machine, and it does store the certificates under `/etc/ssl/certs`, so that is working as expected.

Unfortunately, I haven't found out what causes the issue, but I seems to be with `libsoup` and how it's configured in Venom Linux.
